subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/DefaultScriptCompilationHandler.java

```diff
@@ -285,9 +289,21 @@ public void visitClass(ClassNode node) {
                     customVerifier.execute(node);
                     super.visitClass(node);
                 }
-
             };
-            this.resolveVisitor = new GradleResolveVisitor(this, defaultImportPackages, simpleNameToFQN);
+            final GradleResolveVisitor resolveVisitor = new GradleResolveVisitor(this, defaultImportPackages, simpleNameToFQN);
+            this.resolveVisitor = resolveVisitor;
+            this.progressCallback = new ProgressCallback() {
+                @Override
+                public void call(ProcessingUnit context, int phase) throws CompilationFailedException {
+                    if (phase == Phases.CANONICALIZATION) {
+                        Set<String> deprecatedImports = resolveVisitor.getDeprecatedImports();
+                        if (!deprecatedImports.isEmpty()) {
+                            DeprecationLogger.nagUserWith("You're using " + Joiner.on(",").join(deprecatedImports)
```

Comment:
I'd like to adjust the wording a bit to make it clearer what the problem is:

"Using " + deprecatedImports + " from the private org.gradle.util package without an explicit import is deprecated. Please either stop using these classes (recommended) or import them explicitly at the top of your build file (not recommended). The implicit import will be removed in Gradle 5.0"

Reply:
Done and merged.